homeassistant/components/tankerkoenig/sensor.py

```diff
@@ -0,0 +1,170 @@
+        vol.Inclusive(
+            CONF_LATITUDE, "coordinates", "Latitude and longitude must exist together"
+        ): cv.latitude,
+        vol.Inclusive(
+            CONF_LONGITUDE, "coordinates", "Latitude and longitude must exist together"
+        ): cv.longitude,
+        vol.Optional(CONF_RADIUS, DEFAULT_RADIUS): vol.All(
+            cv.positive_int, vol.Range(min=1)
+        ),
+        vol.Optional(CONF_SCAN_INTERVAL, SCAN_INTERVAL): cv.time_period,
+        vol.Optional(CONF_STATIONS, default=[]): vol.All(cv.ensure_list, [cv.string]),
+    }
+)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set the Tankerkoenig sensor platform up."""
+    fuel_types = config.get(CONF_TYPES, FUEL_TYPES)
+    api_key = config.get(CONF_API_KEY)
+    latitude = config.get(CONF_LATITUDE, hass.config.latitude)
+    longitude = config.get(CONF_LONGITUDE, hass.config.longitude)
+    radius = config.get(CONF_RADIUS, DEFAULT_RADIUS)
+    additional_stations = config.get(CONF_STATIONS, [])
+
+    master = None
+    entities = []
+    _LOGGER.debug("Fetching data for (%s,%s) rad: %s", latitude, longitude, radius)
+    try:
+        data = pytankerkoenig.getNearbyStations(
+            api_key, latitude, longitude, radius, "all", "dist"
+        )
+    except pytankerkoenig.customException as err:
+        data = {"ok": False, "message": err, "exception": True}
+    _LOGGER.debug("Received data: %s", data)
+    if not data["ok"]:
+        _LOGGER.error("Error fetching data from tankerkoenig.de:\n%s", data["message"])
+        _LOGGER.error("Could not setup sensors")
+        return
+
+    if len(data["stations"]) <= 0:
+        _LOGGER.warning("Could not find any station in range")
+    else:
+        for station in data["stations"]:
+            master = add_station(station, api_key, fuel_types, entities, master)
+
+    for additional_station_id in additional_stations:
+        try:
+            additional_station_data = pytankerkoenig.getStationData(
+                api_key, additional_station_id
+            )
+        except pytankerkoenig.customException as err:
+            additional_station_data = {"ok": False, "message": err, "exception": True}
+        if not additional_station_data["ok"]:
+            _LOGGER.warning(
+                "Error when adding station %s: %s",
+                additional_station_id,
+                additional_station_data["message"],
+            )
+        else:
+            master = add_station(
+                additional_station_data["station"],
+                api_key,
+                fuel_types,
+                entities,
+                master,
+            )
+
+    async_add_entities(entities, True)
+
+
+def add_station(station, api_key, fuel_types, entities, master):
+    """Add fuel station to the entity list."""
+    _LOGGER.debug(
+        "Add_station called for station: %s and fuel types: %s", station, fuel_types
+    )
+    station_id = station["id"]
```

Comment:
use this in the `else` clause, it is not needed before.

Reply:
Even better, I don't need the variable at all. Thanks! It was probably a remnant of my debugging code, where I had logs everywhere.